docs/docs/reference-react-component.md

```diff
@@ -0,0 +1,261 @@
+
+### `componentWillReceiveProps(nextProps)`
+
+`componentWillReceiveProps()` is invoked before a mounted component receives new props. If you need to update the state in response to prop changes (for example, to reset it), you may compare `this.props` and `nextProps` and perform state transitions using `this.setState()` in this method.
+
+Note that React may call this method even if the props have not changed, so make sure to compare the current and next values if you only want to handle changes. This may occur when the parent component causes your component to re-render.
+
+`componentWillReceiveProps()` is not invoked if you just call `this.setState()`
+
+### `componentWillUnmount()`
+
+`componentWillUnmount()` is invoked immediately before a component is unmounted and destroyed. Perform any necessary cleanup in this method, such as invalidating timers, canceling network requests, or cleaning up any DOM elements that were created in `componentDidMount`
+
+### `componentWillUpdate(nextProps, nextState)`
+
+`componentWillUpdate()` is invoked immediately before rendering when new props or state are being received. Use this as an opportunity to perform preparation before an update occurs. This method is not called for the initial render.
+
+Note that you cannot call `this.setState()` here. If you need to update state in response to a prop change, use `componentWillReceiveProps()` instead.
+
+> Note
+>
+> `componentWillUpdate()` will not be invoked if [`shouldComponentUpdate()`](#shouldcomponentupdatenextprops-nextstate) returns false.
+
+
+### `defaultProps`
+
+`defaultProps` can be defined as a property on the component class itself, to set the default props for the class. This is used for undefined props, but not for null props. For example:
+
+```js
+class CustomButton extends React.Component {
+  // ...
+}
+
+CustomButton.defaultProps = {
+  color: 'blue'
+};
+```
+
+If `props.color` is not provided, it will be set by default to `'blue'`:
+
+```js
+  render() {
+    return <CustomButton /> ; // props.color will be set to blue
+  }
+```
+
+If `props.color` is set to null, it will remain null:
+
+```js
+  render() {
+    return <CustomButton color={null} /> ; // props.color will remain null
+  }
+```
+
+### `displayName`
+
+The `displayName` string is used in debugging messages. JSX sets this value automatically; see [JSX in Depth](/react/docs/jsx-in-depth.html).
+
+### `forceUpdate(callback)`
+
+`forceUpdate()` can be invoked on any mounted component when you know that some deeper aspect of the component's state has changed without using `this.setState()`
+
+By default, when your component's state or props change, your component will re-render. However, if these change implicitly (eg: data deep within an object changes without changing the object itself) or if your `render()` method depends on some other data, you can tell React that the component needs re-rendering by calling `forceUpdate()`
```

Comment:
yeah this paragraph is pretty awkward. In my defense I just copied it from the previous docs :P I split the sentence and cleaned up a bit